### PR TITLE
docs: add proposal process

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
Add the proposal process for contributors.

### What is changed and how it works?
N/A

### Check List 

Tests 
N/A

Code changes
N/A
 
Side effects
N/A

Related changes

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason @winkyao

File: docs/design/README.md [hunk 1]
```diff
@@ -1,12 +1,28 @@
 # TiDB Design Documents
 
-A proposal template: [TEMPLATE.md](./TEMPLATE.md)
+## Why We need Design Documents
```
Comment (CaitinChen) [inline docs/design/README.md]:
```suggestion
## Why We Need Design Documents
```

File: docs/design/README.md [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
 
-## Proposed
+The design document provides a description of the architecture and important details of what we want to do through a high-level. It is the most powerful tool to ensure our work to be correctly performed.
```
Comment (CaitinChen) [inline docs/design/README.md]: What's the meaning of "through a high-level"?

Comment (CaitinChen) [inline docs/design/README.md]:
```suggestion
The design document provides a description of the architecture and important details of what we want to do through a high-level. It is the most powerful tool to ensure that our work is correctly performed.
```

File: docs/design/README.md [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+
+Writing a design document can promote us to think deliberately and gather the knowledge from others to get our job be done better.  An excellent design document is closely related to the success of our task.
```
Comment (CaitinChen) [inline docs/design/README.md]:
```suggestion
Writing a design document can promote us to think deliberately and gather knowledge from others to get our job done better. An excellent design document is closely related to the success of our task.
```

File: docs/design/README.md [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
+
+## Proprosal Process
+
+1. Before starting the design document, please [create a new issue](https://github.com/pingcap/tidb/issues/new/choose) for tracing the process of your design.
```
Comment (CaitinChen) [inline docs/design/README.md]:
```suggestion
1. Before starting to write a design document, please [create a new issue](https://github.com/pingcap/tidb/issues/new/choose) for tracing the process of your design.
```

File: docs/design/README.md [hunk 5]
```diff
@@ -1,12 +1,28 @@
...
+2. Create a new [Google Doc](https://docs.google.com/document/u/0/) to write the design document. Pelase refer to this [proposal template](./TEMPLATE.md).
+3. If the design document is ready to be reviewed, please put the shared link with `can comment` permission in the issue you've created, one of our engineers will follow up and keep updated.
```
Comment (CaitinChen) [inline docs/design/README.md]:
```suggestion
3. If the design document is ready to be reviewed, please put the shared link with the `can comment` permission in the issue you've created, and one of our engineers will follow up this issue and keep it updated.
```

File: docs/design/README.md [hunk 6]
```diff
@@ -1,12 +1,28 @@
...
+4. If needed, we may organize a communications seminar through [Google Hangouts](https://hangouts.google.com/) to discuss your design with you and the interested friends from the community.
+5. When your design is determined, please submit a pull request(PR) to add your new file under this directory, and put the link of your PR in the issue you've created.
```
Comment (CaitinChen) [inline docs/design/README.md]:
```suggestion
5. When your design is finalized, please submit a pull request (PR) to add your new file under this directory, and put the link of your PR in the issue you've created. 
```

File: docs/design/README.md [hunk 7]
```diff
@@ -1,12 +1,28 @@
...
+
+## Proprosal Process
+
+1. Before starting the design document, please [create a new issue](https://github.com/pingcap/tidb/issues/new/choose) for tracing the process of your design.
+2. Create a new [Google Doc](https://docs.google.com/document/u/0/) to write the design document. Pelase refer to this [proposal template](./TEMPLATE.md).
+3. If the design document is ready to be reviewed, please put the shared link with `can comment` permission in the issue you've created, one of our engineers will follow up and keep updated.
+4. If needed, we may organize a communications seminar through [Google Hangouts](https://hangouts.google.com/) to discuss your design with you and the interested friends from the community.
+5. When your design is determined, please submit a pull request(PR) to add your new file under this directory, and put the link of your PR in the issue you've created. 
+6. Once your PR has been merged, please close the old issue.
+7. Start the implementation refer to the proposal, and create a new issue to trace the process.
```
Comment (CaitinChen) [inline docs/design/README.md]:
```suggestion
7. Start the implementation referring to the proposal, and create a new issue to trace the process.
```

**Comment by morgo** [comment]
LGTM

**Pull request merged**